src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1341,14 +1355,14 @@ protected override LocalState ReachableBottomState()
 
         private void EnterParameters()
         {
-            var methodSymbol = _symbol as MethodSymbol;
-            if (methodSymbol is null)
+            if (!(_symbol is MethodSymbol methodSymbol))
             {
                 return;
             }
 
             var methodParameters = methodSymbol.Parameters;
-            var signatureParameters = _useMethodSignatureParameterTypes ? _methodSignatureOpt.Parameters : methodParameters;
+            var signatureParameters = (_useMethodSignatureParameterTypes ? _delegateInvokeMethod : methodSymbol).Parameters;
```

Comment:
>_useMethodSignatureParameterTypes [](start = 39, length = 33)

Is there a guarantee that ```_delegateInvokeMethod``` is not null when ```_useMethodSignatureParameterTypes``` is true. Constructor used to ensure that before, but no longer does this, and I think there was another place in the code where ```_delegateInvokeMethod``` was explicitly checked for null in addition to ```_useMethodSignatureParameterTypes```.

Reply:
Added assert back.

---
In reply to: [330704941](https://github.com/dotnet/roslyn/pull/38769#discussion_r330704941) [](ancestors = 330704941)